sql/parse_tree_nodes.h

```diff
@@ -1161,20 +1161,33 @@ class PT_option_value_no_option_type_password_for
   virtual bool contextualize(Parse_context *pc);
 };
 
-class PT_option_value_type : public Parse_tree_node {
-  typedef Parse_tree_node super;
-
+class PT_option_type_with_thread_id : public Parse_tree_node {
```

Comment:
I seem to have lost some review comments.

It seems thread_id could be pushed into PT_option_value_type directly and the constructor can support an extra argument for thread_id (enum_var_type, ulong, PT_option_value_following_option_type*)

Then the grammer rule for option_value could be modified to:

```
option_value:
          option_type option_value_following_option_type
          {
            $$= NEW_PTN PT_option_value_type($1, 0, $2);
          }
          | SESSION_SYM opt_var_tid option_value_following_option_type
          {
            $$= NEW_PTN PT_option_value_type(OPT_SESSION, $2, $3);
          }        
          | LOCAL_SYM opt_var_tid option_value_following_option_type
          {
            $$= NEW_PTN PT_option_value_type(OPT_SESSION, $2, $3);
          }
        | option_value_no_option_type { $$= $1; }
        ;
```

It seems then the rest of the grammer rules wouldn't have to change.

Reply:
`option_type_with_thread_id:` is used for both `option_value:` and `start_option_value_list_following_option_type:`. 
IMHO merging `option_value:` with `option_type_with_thread_id` will make everything more complex.

Reply:
Makes sense. The grammer rules for the `start_option_value_list:` case would need to be extended to support the cases where option_type is SESSION,  but disabled for the the SESSION TRANSACTION subset, which is probably more complicated.